en/16_Multiple_Objects.adoc

```diff
@@ -0,0 +1,324 @@
+        // Apply continuous rotation to the object
+        gameObject.rotation.y += 0.001f; // Slow rotation around Y axis
+
+        // Get the model matrix for this object
+        glm::mat4 initialRotation = glm::rotate(glm::mat4(1.0f), glm::radians(-90.0f), glm::vec3(1.0f, 0.0f, 0.0f));
+        glm::mat4 model = gameObject.getModelMatrix() * initialRotation;
+
+        // Create and update the UBO
+        UniformBufferObject ubo{
+            .model = model,
+            .view = view,
+            .proj = proj
+        };
+
+        // Copy the UBO data to the mapped memory
+        memcpy(gameObject.uniformBuffersMapped[currentFrame], &ubo, sizeof(ubo));
+    }
+}
+----
+
+Note that we're sharing the view and projection matrices across all objects, but each object has its own model matrix.
+
+=== Modify the Command Buffer Recording
+
+Finally, we'll update the command buffer recording to draw each object:
+
+[,c{pp}]
+----
+void recordCommandBuffer(uint32_t imageIndex) {
+    // ... (beginning of the method remains the same)
+
+    // Bind vertex and index buffers (shared by all objects)
+    commandBuffers[currentFrame].bindVertexBuffers(0, *vertexBuffer, {0});
+    commandBuffers[currentFrame].bindIndexBuffer(*indexBuffer, 0, vk::IndexType::eUint32);
+
+    // Draw each object with its own descriptor set
+    for (const auto& gameObject : gameObjects) {
+        // Bind the descriptor set for this object
+        commandBuffers[currentFrame].bindDescriptorSets(
+            vk::PipelineBindPoint::eGraphics,
+            *pipelineLayout,
+            0,
+            *gameObject.descriptorSets[currentFrame],
+            nullptr
+        );
+
+        // Draw the object
+        commandBuffers[currentFrame].drawIndexed(indices.size(), 1, 0, 0, 0);
+    }
+
+    // ... (end of the method remains the same)
+}
+----
+
+== Performance Considerations
+
+When rendering multiple objects, keep these performance considerations in mind:
+
+1. *Minimize state changes*: Group objects by material/texture to reduce binding changes.
+2. *Use instancing* for many identical objects (not covered in this tutorial).
+3. *Consider push constants* for small, frequently changing data instead of uniform buffers.
+4. *Batch draw calls* where possible to reduce CPU overhead.
+5. *Use indirect drawing* for large numbers of objects (not covered here).
+
+== Conclusion
+
+You've now learned how to render multiple objects in Vulkan by:
+1. Creating a structure to hold per-object data
+2. Duplicating the necessary resources (uniform buffers, descriptor sets) for each object
+3. Sharing resources that can be reused (vertex/index buffers, pipeline, textures)
+4. Updating the rendering loop to draw each object with its own transformation
+
+This approach gives you the flexibility to position, rotate, and scale objects independently while maintaining good performance by sharing resources where appropriate.
+
+In a real-world application, you might extend this system with:
```

Comment:
This needs an additional line break between before the list, as the list is otherwise not rendered as such.